core/conf/config_test.go

```diff
@@ -0,0 +1,237 @@
+package conf
+
+import (
+	"testing"
+
+	"github.com/0chain/gosdk/core/conf/mocks"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoadConfig(t *testing.T) {
+
+	tests := []struct {
+		name        string
+		exceptedErr error
+
+		setup func(*testing.T) Reader
+		run   func(*require.Assertions, Config)
+	}{
+		{
+			name:        "Test_Config_Invalid_BlockWorker",
+			exceptedErr: ErrInvalidValue,
+			setup: func(t *testing.T) Reader {
+
+				reader := &mocks.Reader{}
+				reader.On("GetString", "block_worker").Return("")
+				reader.On("GetInt", "min_submit").Return(0)
+				reader.On("GetInt", "min_confirmation").Return(0)
+				reader.On("GetInt", "max_txn_query").Return(0)
+				reader.On("GetInt", "query_sleep_time").Return(0)
+				reader.On("GetInt", "confirmation_chain_length").Return(0)
+				reader.On("GetStringSlice", "preferred_blobbers").Return(nil)
+				reader.On("GetString", "signature_scheme").Return("")
+				reader.On("GetString", "chain_id").Return("")
+
+				return reader
+			},
+			run: func(r *require.Assertions, cfg Config) {
+
+			},
+		},
+		{
+			name: "Test_Config_BlockWorker",
+
+			setup: func(t *testing.T) Reader {
+
+				reader := &mocks.Reader{}
```

Comment:
this block seems same for every test, pls code it once and just call it from here, to avoid duplications

Reply:
It must be different instance. Because they have different return values for mock. 

mock.On is impossible to overwrite last mock. We has to create different instance for each test.

Reply:
Yes, you need to configure the `mock.On`s each test, but you can do it in a function and set that function in each test.
```go
setupMyMockDotOns =func(t *testing.T) Reader {repeated code; return reader}
```
```
		{
			name: "Test_Config_BlockWorker",
			setup: setupMyMockDotOns
```

Reply:
for `LoadConfig` , it will read each mock method. so it is better to setup mock object for each test with different values.  Just some values are same as 0 , but not for all.

I would suggest to leave it there. It is more clear for reading and understanding. You can have a try on your local

Reply:
They have different values .eg

- Test_Config_Min_Submit_Greater_Than_100  : `reader.On("GetInt", "min_submit").Return(101)`
- Test_Config_Min_Confirmation_Greater_100 : `reader.On("GetInt", "min_confirmation").Return(101)`



Anyway, I have got it refactored as you like